_posts/2017-01-17-Oredev2016.md

```diff
@@ -0,0 +1,695 @@
+* ports
+    * The port the service will listen to. When you want to expose your port use: "5432:5432"
+* build
+    * This will build the Dockerfile inside the same directory
+* networks as property
+    * This be the network where you want to be assigned to
+* networks
+    * This will create one or multiple networks
+
+When you want to add an eureka service, follow the above steps and configure it inside the docker-compose file.
+It is important that every container you want to communicate in between stays under the same network.
+
+<a name="docker" />
+
+# Secure systems with Docker - Emil Kvarnhammer
+
+<img class="float-image p-image" src="{{ '/img/oredev/emil.jpg' | prepend: site.baseurl }}" alt="Emil Kvarnhammer" style="width: 100px"/>
+
+[Emil Kvarnhammer](http://www.twitter.com/emilkvarnhammar) is a Security Software Engineer at TrueSec,
+a leading-edge company in IT security and development.
+He’s been involved in several security-critical projects,
+developing applications and components that are used by millions of users on a regular basis.
+Emil has found severe security vulnerabilities in Apple OS X. His recent focus has been secure systems using Docker and Amazon EC2.
+
+
+When entering the world of cloud and distributed systems, security can be challenging for developers and devops.
+Docker is a part of this world and Emil talks about Docker in a security perspective.
+
+When looking it in a secure way, what are the input sources?
+Where does the base image come from?
+If we look closer we can see that it retrieves it's images from Docker Hub.
+In addition to that, more complexity comes in when we add new packages on top of the base image which will be retrieved from the package manager server.
+Finally we have the application package that can come from a CI server and if it is a fat jar, it can consists of different dependencies through maven or some kind.
+
+
+<p style="text-align: center;">
+  <img style="max-width:640px;" alt="Hackers perspective" src="/img/oredev/hackers.png">
+</p>
+
+
+The base image is retrieved from Docker Hub every time a docker image is being updated.
+A hacker could hijack the image and squeeze in malicious software without anyone noticing.
+Same goes for the package manager server.
+
+What could an attacker do if he gets control of the channel between the docker image and
+the host operating system?
+The place where the image gets instantiated or the configuration related to the instantiated image.
+An attacker could infect the image by his own configuration or modify the orchestration tool.
+
+The most important issue is when the attacker gets control of code execution within our application.
+Emil says that it happens quite often and if it isn't in your code, it is in the third party libraries or the package manager server you're using.
+
+Now you might think that because the attacker is in a docker container that he can't break out.
```

Comment:
Now you might think that, because the attacker is in a docker container, he can't break out.